pkg/utils/backoff/retry.go

```diff
@@ -0,0 +1,109 @@
+package backoff
+
+import (
+	"context"
+	"time"
+)
+
+// DefaultMaxElapsedTime sets a default limit for the total retry duration.
+const DefaultMaxElapsedTime = 1 * time.Minute
+
+// Operation is a function that attempts an operation and may be retried.
+type Operation[T any] func() (T, error)
```

Comment:
Is this important to define as a separate named type? It requires callers to cast in some situations